### PR TITLE
Upgrade Checkstyle to 9.3

### PR DESCRIPTION
**Pull request by pugnascotia**
Checkstyle 9.3 has been released, and contains a fix for the memory problems we saw in previous versions. Upgrade to this version. 

Note that the IDEA config has also been updated, but the plugin doesn't yet have 9.3 as an option according to the release notes. This shouldn't be a problem.

### CODE REVIEW

File: .idea/checkstyle-idea.xml [hunk 1]
```diff
@@ -3,17 +3,17 @@
   <component name="CheckStyle-IDEA">
     <option name="configuration">
       <map>
-        <entry key="active-configuration" value="PROJECT_RELATIVE:$PROJECT_DIR$/checkstyle_ide.xml:Elasticsearch" />
-        <entry key="checkstyle-version" value="8.42" />
+        <entry key="active-configuration" value="PROJECT_RELATIVE:/Users/rory/src/elastic/elasticsearch/checkstyle_ide.xml:Elasticsearch" />
```
Comment (mark-vieira) [inline .idea/checkstyle-idea.xml:6]: Looks like IntelliJ snuck this change in. We should still be using `$PROJECT_DIR$` here.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Pull request merged**